libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -39,8 +40,20 @@ public MotionSensor(Context appCtx) {
         // If no TYPE_SIGNIFICANT_MOTION is available, use alternate means to sense motion
         if (motionSensor == null) {
             motionSensor = new LegacyMotionSensor(mAppContext);
-            AppGlobals.guiLogInfo("Device has legacy motion sensor.");
         }
+        setTypeFromPrefs();
+    }
+
+    private void setTypeFromPrefs() {
+        if (Prefs.getInstance(mAppContext).getIsMotionSensorTypeSignificant()) {
+            motionSensor = SignificantMotionSensor.getSensor(mAppContext);
+        }
+    }
+
+    // Call when the scanning has stopped (not when paused).
+    // This class can use this event to update its state
+    public void scannerFullyStopped() {
+        setTypeFromPrefs();
```

Comment:
MotionSensor needs an event to trigger when to check if the Prefs have changed. The Power Options screen stops and restarts the scanning when options are changed.